### PR TITLE
714 raster refactor

### PR DESCRIPTION
**Pull request by gantian127**

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

File: hs_geo_raster_resource/models.py [hunk 1]
```diff
@@ -60,23 +60,25 @@ def create(cls, **kwargs):
 
     @classmethod
     def update(cls, element_id, **kwargs):
```
Comment (ironfroggy) [inline hs_geo_raster_resource/models.py]: Add a docstring and explain the new behavior.

File: hs_geo_raster_resource/models.py [hunk 2]
```diff
@@ -50,9 +50,9 @@ def create(cls, **kwargs):
                               if k in ('units', 'northlimit', 'eastlimit', 'southlimit', 'westlimit', 'projection')}
```
Comment (ironfroggy) [inline hs_geo_raster_resource/models.py]: add a docstring to create() and explain why this class needs to override the create() from its base

**Comment by ironfroggy** [comment]
You've made some model changes but not checked in any migrations. Please include the migrations.

**Comment by gantian127** [comment]
@ironfroggy Would you please review the updated code? Thank you.

File: hs_geo_raster_resource/receivers.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def raster_pre_create_resource_trigger(sender, **kwargs):
                 ('columns', res_md_dict['cell_and_band_info']['columns']),
                 ('cellSizeXValue', res_md_dict['cell_and_band_info']['cellSizeXValue']),
                 ('cellSizeYValue', res_md_dict['cell_and_band_info']['cellSizeYValue']),
-                ('cellSizeUnit', res_md_dict['cell_and_band_info']['cellSizeUnit']),
+                # ('cellSizeUnit', res_md_dict['cell_and_band_info']['cellSizeUnit']),
```
Comment (pkdash) [inline hs_geo_raster_resource/receivers.py]: should remove commented code

Comment (ironfroggy) [inline hs_geo_raster_resource/receivers.py]: if this isn't needed remove it

**Comment by ironfroggy** [comment]
+1 if comment is addressed

File: hs_geo_raster_resource/receivers.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 # to populate create-resource.html template page
 @receiver(pre_create_resource, sender=RasterResource)
 def raster_pre_create_resource_trigger(sender, **kwargs):
-    if(sender is RasterResource):
+    if sender is RasterResource:
```
Comment (pkdash) [inline hs_geo_raster_resource/receivers.py]: If statement is not needed as the decorator already has the restriction (sender=RasterResource)

File: hs_geo_raster_resource/receivers.py [hunk 5]
```diff
@@ -82,27 +78,23 @@ def raster_pre_create_resource_trigger(sender, **kwargs):
             band_dict['comment'] = ''
             metadata.append({'BandInformation': band_dict})
 
+
 @receiver(pre_add_files_to_resource, sender=RasterResource)
 def raster_pre_add_files_to_resource_trigger(sender, **kwargs):
-    if(sender is RasterResource):
+    if sender is RasterResource:
```
Comment (pkdash) [inline hs_geo_raster_resource/receivers.py]: Same - if statement is not needed

File: hs_geo_raster_resource/receivers.py [hunk 6]
```diff
@@ -125,32 +117,16 @@ def raster_pre_add_files_to_resource_trigger(sender, **kwargs):
             for i in range(bcount):
                 res.metadata.create_element('BandInformation', name='Band_' + str(i+1), variableName='Unnamed', variableUnit='Unnamed', method='', comment='')
 
+
 @receiver(pre_delete_file_from_resource, sender=RasterResource)
 def raster_pre_delete_file_from_resource_trigger(sender, **kwargs):
-    if(sender is RasterResource):
+    if sender is RasterResource:
```
Comment (pkdash) [inline hs_geo_raster_resource/receivers.py]: same - if statement is not needed

**Comment by pkdash** [comment]
+1

**Pull request merged**